frontend/src/components/PlanHome/PlanHome.tsx

```diff
@@ -30,28 +36,136 @@ const planProgressCalculator = (currentStage: PlanStage): number[] => {
+  const [progressPercentage, taskCompleted] = planProgressCalculator(PlanStage.DELIVERY)
+  const [showInviteModal, setShowInviteModal] = useState<boolean>(false)
+  const [selectedContacts, setSelectedContacts] = useState<Contact[] | []>([]);
+  const [message, setMessage] = useState<string | undefined>()
+  const [displayMessage, setDisplayMessage] = useState<string>('');
   const { id } = useParams();
+  const {
+    planData,
+    contactData
+  } = fetchPlanAndContactsData(id as string)
+  useEffect(() => {
+    const handleClickOutsideOfModal = (event: any) => {
+      if (showInviteModal && event.target.closest('.clickOutsideOfModal') === null) {
+        handleDiscardModal()
+      }
+    }
+    if (showInviteModal) {
+      document.addEventListener('mousedown', handleClickOutsideOfModal)
+    } else {
+      document.removeEventListener('mousedown', handleClickOutsideOfModal)
+    }
+  }, [showInviteModal])
 
-  // TODO: Make the component load data from backend
-  const [, status] = fetchPlanData(id as string)
+  const handleContactSelect = (singleContact: Contact): void => {
+    setSelectedContacts((prev: Contact[]) => [...prev, singleContact])
+  }
 
-  if (status === LoadStatus.LOADING || status === LoadStatus.NOT_LOADED) {
-    return (
-      <div>
-        <p>loading plan</p>
-      </div>
-    )
-  } else if (status === LoadStatus.FAILED) {
-    return (
-      <div>
-        <p>couldn't fetch plan</p>
-      </div>
+  const handleRemoveContactTag = (contactToBeRemoved: Contact) => {
+    const filteredContacts = selectedContacts.filter(
+      (currentContacts: Contact) =>
+        currentContacts.id !== contactToBeRemoved.id
     )
+
+    setSelectedContacts(filteredContacts)
   }
 
-  const [progressPercentage, taskCompleted] = planProgressCalculator(PlanStage.DELIVERY)
+  const handleDiscardModal = () => {
+    setSelectedContacts([])
+    setShowInviteModal(false)
+    setMessage(undefined)
+    removeModal()
+  }
+
+  const handleSubmit = async (): Promise<void> => {
+    if (selectedContacts.length === 0) {
+      handleDisplayMessage('Add at least 1 contact')
+      return
+    }
+
+    if (message === undefined) {
+      handleDisplayMessage('Add a message')
+      return
+    }
+
+    try {
+      await axios.post('/api/inviteContacts', {
+        selectedContacts,
+        message,
+        planID: id
+      }, {
+        withCredentials: true
+      })
+      // TODO: add 'pending' users to the member list
+      handleDisplayMessage('Sent successfully')
+    } catch (error) {
+      if (error instanceof AxiosError) {
+        handleDisplayMessage('Fail to send. Please retry.')
+        // TODO: Add error handling for mutliple simulatenaous invites, aka spamming the send button
+        console.error(error?.response?.data)
+      }
+    }
+  }
+
+  const handleDisplayMessage = (message: string): void => {
+    if (displayMessage.length > 0) return
```

Comment:
is this conditional correct?

Reply:
if so, the intent of the logic isn't clear, so let's use more descriptive fn naming or comments